app/lib/discovery_filter_validator.rb

```diff
@@ -0,0 +1,19 @@
+class DiscoveryFilterValidator < ActiveModel::Validator
+  # Users can assign a search filter to a discovered host that is valid
+  # only for the Host model, but not for the Host::Discovered model.
+  # For example, the 'hostgroup' field is valid for Host, but not for Host::Discovered.
+  def validate(record)
+    return if record.search.blank?
+
+    has_permission = record.permissions.find { |perm| perm.name == 'view_discovered_hosts' }
+
+    return unless has_permission
+
+    begin
+      Host::Discovered.search_for(record.search)
+    rescue ScopedSearch::QueryNotSupported => _e
+      msg = N_("Search filter is invalid: Field 'search' is not recognized for Discovered hosts.")
```

Comment:
@Lennonka text review please. Looking at it, the message right now doesn't make sense, the field `search` is recognized for Discovered hosts, but its value is wrong. So yeah, this needs to be updated.

Reply:
Can we tell the user specifically what in the value is invalid?

Reply:
For example:
```suggestion
      msg = N_("Search query is invalid: 'hostgroup' is not recognized in search for discovered hosts.")
```
Not sure if I understand it correctly.